yoti-sdk-api/src/main/java/com/yoti/api/client/identity/policy/Policy.java

```diff
@@ -0,0 +1,282 @@
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+import com.yoti.api.attributes.AttributeConstants;
+import com.yoti.api.client.identity.constraint.Constraint;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Policy {
+
+    @JsonProperty(Property.WANTED)
+    private final Collection<WantedAttribute> wantedAttributes;
+
+    @JsonProperty(Property.WANTED_AUTH_TYPES)
+    private final Set<Integer> wantedAuthTypes;
+
+    @JsonProperty(Property.WANTED_REMEMBER_ME)
+    private final boolean wantedRememberMe;
+
+    @JsonProperty(Property.WANTED_REMEMBER_ME_OPTIONAL)
+    private final boolean wantedRememberMeOptional;
+
+    @JsonProperty(Property.IDENTITY_PROFILE_REQUIREMENTS)
+    private final Object identityProfile;
+
+    private Policy(Builder builder) {
+        wantedAttributes = builder.wantedAttributes.values();
+        wantedAuthTypes = builder.wantedAuthTypes;
+        wantedRememberMe = builder.wantedRememberMe;
+        wantedRememberMeOptional = builder.wantedRememberMeOptional;
+        identityProfile = builder.identityProfile;
+    }
+
+    public Collection<WantedAttribute> getWantedAttributes() {
+        return wantedAttributes;
+    }
+
+    public Set<Integer> getWantedAuthTypes() {
+        return wantedAuthTypes;
+    }
+
+    public boolean isWantedRememberMe() {
+        return wantedRememberMe;
+    }
+
+    public boolean isWantedRememberMeOptional() {
+        return wantedRememberMeOptional;
+    }
+
+    public Object getIdentityProfile() {
+        return identityProfile;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+
+        private static final int SELFIE_AUTH_TYPE = 1;
+        private static final int PIN_AUTH_TYPE = 2;
+
+        private final Map<String, WantedAttribute> wantedAttributes;
+        private final Set<Integer> wantedAuthTypes;
+
+        private boolean wantedRememberMe;
+        private boolean wantedRememberMeOptional;
+        private Object identityProfile;
+
+        private Builder() {
+            wantedAttributes = new HashMap<>();
+            wantedAuthTypes = new HashSet<>();
+        }
+
+        public Builder withWantedAttribute(WantedAttribute wanted) {
+            String key = Optional.ofNullable(wanted.getDerivation()).orElseGet(wanted::getName);
+
+            if (wanted.hasConstraints()) {
+                if(wantedAttributes.values().stream()
+                        .map(WantedAttribute::getConstraints)
+                        .noneMatch(c -> c.equals(wanted.getConstraints()))) {
+                    key += "-" + UUID.randomUUID();
+                    wantedAttributes.put(key, wanted);
+                }
+
+                return this;
+            }
+
+            wantedAttributes.put(key, wanted);
+            return this;
```

Comment:
Logic of this method seems a bit complicated... wouldn't this result in the same outcome (and be simpler to grok?):
```java
String key = Optional.ofNullable(wanted.getDerivation()).orElseGet(wanted::getName);

if (!wanted.hasConstraints()) {
    wantedAttributes.put(key, wanted);
    return this;
}

if (hasMultipleConstraints(wantedAttributes)) {
    wantedAttributes.put(key + "-" + UUID.randomUUID(), wanted);
}

return this;
```